ls1intum/Artemis
# `Programming exercises`: Store the used branch in the database

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises

### Motivation and Context


Retrieving the default branch from the VCS for many git operations is not performant and can slow down the system.

### Description

This PR stores the default Branch for the used repositories in the database of Artemis and only retrieves them from the VCS if necessary.
This PR contains a database change. Make surre to delete all newly created programming exercises and participations after testing!

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise

1. Log in to Artemis
2. Create programming exercise
3. Make sure that everything is still working in the programming exercise:
- Online Editor for Students
- - Online Editor for Instructors
- Cloning to the local IDE, pushing and. receiving feedback
- Importing (Especially If you still have an old exercise with "master" as default branch)
- ...

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
